realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp

```diff
@@ -130,8 +132,17 @@ class JniConfigWrapper {
             user = SyncManager::shared().get_user(user_identity, refresh_token,
                                                   realm::util::Optional<std::string>(realm_auth_url));
         }
+
+        util::Optional<std::string> ssl_trust_certificate_path = util::none;
+        if (sync_ssl_trust_certificate_path) {
+            ssl_trust_certificate_path =
+                realm::util::Optional<std::string>(JStringAccessor(env, sync_ssl_trust_certificate_path));
+        }
         m_config.sync_config = std::make_shared<SyncConfig>(SyncConfig{
-            user, realm_url, SyncSessionStopPolicy::Immediately, std::move(bind_handler), std::move(error_handler)});
+            user, realm_url, SyncSessionStopPolicy::Immediately, std::move(bind_handler), std::move(error_handler),
+            nullptr, util::none, sync_client_validate_ssl, ssl_trust_certificate_path});
```

Comment:
yes setting the `realm_encryption_key` here should fix https://github.com/realm/realm-java/issues/4561 as well